types/crypto-js/index.d.ts

```diff
@@ -1,171 +1,1731 @@
+                 *
+                 * @example
+                 *
+                 *     var HmacSHA256 = CryptoJS.lib.Hasher._createHmacHelper(CryptoJS.algo.SHA256);
+                 */
+                _createHmacHelper(hasher: HasherStatic): HmacHasherHelper;
+            };
+
+            const Cipher: {
+                /**
+                 * Creates shortcut functions to a cipher's object interface.
+                 *
+                 * @param cipher The cipher to create a helper for.
+                 *
+                 * @return An object with encrypt and decrypt shortcut functions.
+                 *
+                 * @example
+                 *
+                 *     var AES = CryptoJS.lib.Cipher._createHelper(CryptoJS.algo.AES);
+                 */
+                _createHelper(cipher: Cipher): CipherHelper;
+            };
+
+            /**
+             * A collection of cipher parameters.
+             */
+            interface CipherParams {
+                /**
+                 * The raw ciphertext.
+                 */
+                ciphertext: WordArray;
+                /**
+                 * The key to this ciphertext.
+                 */
+                key: WordArray;
+                /**
+                 * The IV used in the ciphering operation.
+                 */
+                iv: WordArray;
+                /**
+                 * The salt used with a key derivation function.
+                 */
+                salt: WordArray;
+                /**
+                 * The cipher algorithm.
+                 */
+                algorithm: CipherStatic;
+                /**
+                 * The block mode used in the ciphering operation.
+                 */
+                mode: Mode;
+                /**
+                 * The padding scheme used in the ciphering operation.
+                 */
+                padding: Padding;
+                /**
+                 * The block size of the cipher.
+                 */
+                blockSize: number;
+                /**
+                 * The default formatting strategy to convert this cipher params object to a string.
+                 */
+                formatter: Format;
+                /**
+                 * Converts this cipher params object to a string.
+                 *
+                 * @param formatter (Optional) The formatting strategy to use.
+                 *
+                 * @return The stringified cipher params.
+                 *
+                 * @throws Error If neither the formatter nor the default formatter is set.
+                 *
+                 * @example
+                 *
+                 *     var string = cipherParams + '';
+                 *     var string = cipherParams.toString();
+                 *     var string = cipherParams.toString(CryptoJS.format.OpenSSL);
+                 */
+                toString(formatter?: Format): string;
```

Comment:
Please double check if this can accept Encoder.

Reply:
Not really. `Encoder` can only stringify `WordArray`, not `CipherParams`.